Add initial pages for release documentation and roadmap of architecture support (#194)
ROCm/TheRock: RELEASES.md

# Releases

This is a quick overview of how to consume our current build and release [artifacts](docs/development/artifacts.md).

## Current state

Currently, we produce build artifacts as part of our CI workflows ([example](.github/workflows/build_linux_packages.yml)) as well as part of our release. As the project is still not ready for production (see [ROADMAP](ROADMAP.md)) this doc assumes you are already familiar with how to use ROCm. If not - you should not start here, please start at [ROCm](https://github.com/ROCm/ROCm).

> ## Installing from Tarballs

Reviewer: nit: use consistent capitalization in headings. I like the guidance from https://developers.google.com/style/capitalization#capitalization-in-titles-and-headings to use `Sentence case` rather than `Title Case`

Author: Ack

The quickest way to "install" our artifacts is to leverage the `build_tools/fileset_tool.py artifact-flatten` command. You will need to have a [checkout](README.md#Checkout-Sources) to leverage this tool and a Python environment.

> Here's a quick example assuming you copied the all the tar files into `${BUILD_ARTIFACTS_DIR}` to "install" TheRock into `${BUILD_ARTIFACTS_DIR}/output_dir`

Reviewer: What does "installing" do?

It looks like this only extracts files onto disk, and the `rocminfo` test command below is called directly from that location. Should a user set some environment variables to point their applications at the installation? Or add that location to their `PATH`? Presumably `fileset_tool.py artifact-flatten` does not make such changes itself.

Maybe we should have an installation script that the user can `source therock_rocm_install.sh` or something to help with that.

```bash
echo "Unpacking artifacts"
pushd "${BUILD_ARTIFACTS_DIR}"
mkdir output_dir
python build_tools/fileset_tool.py artifact-flatten *.tar.xz -o output_dir --verbose
popd
```

## Testing your installation

The quickest way is to run `rocminfo`

```bash
echo "Running rocminfo"
pushd "${BUILD_ARTIFACTS_DIR}"
./output_dir/bin/rocminfo
popd

> ```
> ## Where to Get Artifacts

Reviewer: nit: newline after code block and before section header

````suggestion
```

## Where to Get Artifacts
````

Markdownlint has rules for both of those:

* https://github.com/DavidAnson/markdownlint/blob/main/doc/md022.md

    > Rationale: Aside from aesthetic reasons, some parsers, including kramdown, will not parse headings that don't have a blank line before, and will parse them as regular text.

* https://github.com/DavidAnson/markdownlint/blob/main/doc/md031.md

    > Rationale: Aside from aesthetic reasons, some parsers, including kramdown, will not parse fenced code blocks that don't have blank lines before and after them.

Author: Yeah sorry, I've been having a challenging time getting a md formatter that's consistent without our presubmit checks in vscode so I'm cleaning these up as I upload. If you have suggestions for plugins other than just running the markdown linter locally let me know.

Reviewer: `pre-commit run` or `pre-commit install` ;)

That's the whole point of pre-commit - it handles installs and versioning of linters for you: https://pre-commit.com/

Reviewer: https://marketplace.visualstudio.com/items?itemName=elagil.pre-commit-helper might work for pre-commit as a "format on save" tool but I haven't tried it myself. I typically edit using individually installed formatters (e.g. clang-format, black) then have pre-commit run when I commit via the git hook (`pre-commit install`)

Author: Ooo!!! Installing.

> - [Releases](releases): Our releases page has the latest "developer" release of our tarball artifacts and source code.

Reviewer: This link is a 404. Relative links from source code only point to other source files, not parts of GitHub.
```suggestion
- [Releases](https://github.com/ROCm/TheRock/releases): Our releases page has the latest "developer" release of our tarball artifacts and source code.
```

Author: Thanks missed that one!

> - [Releases](https://github.com/ROCm/TheRock/releases): Our releases page has the latest "developer" release of our tarball artifacts and source code.

Reviewer: Ah, that release is already in the right format, no need to run `fileset_tool` at all?

```bash
# Extract
mkdir output_dir
tar -xf *.tar.gz -C output_dir

# Test
./output_dir/bin/rocminfo
```

> - [Releases](https://github.com/ROCm/TheRock/releases): Our releases page has the latest "developer" release of our tarball artifacts and source code.

Author: Thanks for testing this out - I had assumed wrongly that the releases and CI artifacts had the same format will need to update it when I get a chance to test both later today

> - [Releases](https://github.com/ROCm/TheRock/releases): Our releases page has the latest "developer" release of our tarball artifacts and source code.

Reviewer: https://github.com/ROCm/TheRock/blob/main/docs/development/artifacts.md

The CI jobs operate on archive slices. One way to use those slices is to assemble them into a release (dep, rpm, wheel, flat archive).

> - [Releases](https://github.com/ROCm/TheRock/releases): Our releases page has the latest "developer" release of our tarball artifacts and source code.

Author: I've separated releases from CI job documentation. I copied what we're doing in CI with flattening artifacts to give you the equivalent of a release.

- [Packages](https://github.com/orgs/ROCm/packages?repo_name=TheRock): We currently publish docker images for LLVM targets we support (as well as a container for our build machines)

> - [CI](https://github.com/ROCm/TheRock/actions/workflows/ci.yml?query=branch%3Amain+is%3Asuccess): Each of our latest green builds has its own artifacts you can leverage. This is the latest and greatest! We will eventually support a nightly release that is at a higher quality bar than CI.

Reviewer: Prefer to include a few more characters in the link text so it is easier to click/tap on smaller screens. Could also swap "green" for "passing" or "successful" to avoid the colloquialism

```suggestion
- [Per-commit CI builds](https://github.com/ROCm/TheRock/actions/workflows/ci.yml?query=branch%3Amain+is%3Asuccess): Each of our passing CI builds has its own artifacts you can leverage. This is the latest and greatest! We will eventually support a nightly release that is at a higher quality bar than CI.
```

Author: Ack
